tests/c/dyn_ptradd_multidim.newcg.c

```diff
@@ -14,9 +14,9 @@
 //     --- End aot ---
 //     --- Begin jit-pre-opt ---
 //     ...
-//     %{{16}}: ptr = dyn_ptr_add %{{3}}, %{{_}}, {{64}}
```

Comment:
@ltratt we need to change the register value else the test fails when we turn on `mem2reg`.

Reply:
What does "change the register value" mean? The numbers inside `{{...}}` are for fm, and have no semantic meaning other than "the same name must have the same value each time it is referenced" (please read fm's docs on name matching).

Reply:
Sorry, one of the load is optimised out and everything shifts up so dyn_ptr_add check starts from %{{15}} instead of  %{{16}}. We can instead change the check to do `%{{_}}: ptr = dyn_ptr_add %{{3}}, %{{_}}, {{64}}` instead so if any loads/stores are optimised out we don't need to change the value at `{{}}`?

Reply:
Have you read the fm documentation?

Reply:
I read the fm docs (https://docs.rs/fm/0.3.0/fm/index.html), am not sure what am I missing? If a load is optimised out then wouldn't we have lesser number of numerical names with assigned values? And now `%16` is assigned with different value than we expected?

Reply:
What happens if you revert these lines? Does the test pass or fail?

Reply:
the test fails

Reply:
I'm afraid you'll have to debug why (or explain to me why I'm wrong!) as the changed test seems identical to me. AFAICS the two tests are identical modulo alpha-renaming, which shouldn't affect fm's matching.

Reply:
One thing that I do when I can't figure out why something doesn't match, is I re-paste+amend the lines in question from the actual output. Usually the problem makes itself apparent.

Reply:
@ltratt We have multiple lines with `dyn_ptr_add` in the test (one below the other). Now, if the trace changes such that : 
```
%{{16}}: ptr = dyn_ptr_add %{{3}}, %{{_}}, {{64}}
```
becomes 
```
%{{15}}: ptr = dyn_ptr_add %{{3}}, %{{_}}, {{64}}
```
and 
```
%17: ptr = dyn_ptr_add %{{16}}, %{{_}}, {{16}}
```
becomes
```
%16: ptr = dyn_ptr_add %{{15}}, %{{_}}, {{16}}
```
fwiu the test thinks both `dyn_ptr_add` assignments are same and picks the second assignment to start the check.
So, instead of checking:
```
  -> %15: ptr = dyn_ptr_add %3, %12, 64
       %16: ptr = dyn_ptr_add %15, %13, 16
       %17: ptr = dyn_ptr_add %16, %14, 4
```
it starts the check at:
```
     %15: ptr = dyn_ptr_add %3, %12, 64
 -> %16: ptr = dyn_ptr_add %15, %13, 16
     %17: ptr = dyn_ptr_add %16, %14, 4

```

Reply:
The numbers in the brackets have no semantic meaning as such. You can replace every instance of `{{16}}`, for example, with `{{banana}}` (i.e. alpha rename) and you should have the same test. If that isn't the case, it's a bug (probably in fm).

[Warning: I think your change is just alpha renaming, but maybe I'm misreading it.]

Reply:
It looks like this is a `fm` bug. Should I file a bug report?

Reply:
Yes but you will need to provide a minimal test case in order for me to look at it.